notification/src/utils/error-utils.js

```diff
@@ -21,7 +21,17 @@ function isRecoverableError(err) {
 }
 
 function getErrorCause(err) {
-  if (err instanceof VError) return err.cause()
+  if (err instanceof VError) {
+    const cause = err.cause()
+    return {
+      name: cause.name,
+      message: cause.message,
+      stack: cause.stack,
+      body: cause.body,
+      statusCode: cause.statusCode,
+    }
+  }
+
   return err
```

Comment:
I'm sure that this will return an empty log on cause field (try with throwing a simple Error object and check out it's logs), with that I could see that Verror cause is returning an instance of the js Error object and bunyan  (logging module) can not log it properly, maybe this could narrow the investigation ?

Reply:
I found the logging issue here 😉 

In default the bunyan can serialize an `Error object` on err field of the log context, here the Error object is passed in a field called `cause` so bunyan is not serializing it to stdout 

see details: https://github.com/trentm/node-bunyan#requirements-for-serializers-functions

So overcome the issue a custom serializer for `cause` field could be added to the logger: 

https://github.com/commercetools/commercetools-adyen-integration/blob/0602d6529b579cf3f634feb403f7f3d8d2ddea0c/notification/src/utils/logger.js#L12-L14

to:
````javascript
serializers: { err: bunyan.stdSerializers.err, cause: bunyan.stdSerializers.err },
`````

Another approach might be renaming cause to err.

NOTE: I have not test all type of errors (verror, normal error and network error) so please test it.

Reply:
Thanks for your time. https://github.com/commercetools/commercetools-adyen-integration/pull/816/commits/c64df4f331040a2782d4246630d76a3c141e2104